### PR TITLE
97657: Removed "FromPort" in netpol tests

### LINKED ISSUE #97657: remove or comment `FromPort` in netpol tests
**What would you like to be added**:

Remove or comment `FromPort` from new netpol tests https://github.com/kubernetes/kubernetes/blob/43fcd50d50429e469b501ac7774ad2ffb3235b60/test/e2e/network/netpol/reachability.go#L28

**Why is this needed**:

The `FromPort` field is confusing and undocumented -- we should clarify its role or remove it.

### PR DESCRIPTION
**Pull request by dmffen**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removed unused and misleading field in a test structure

**Which issue(s) this PR fixes**:

Fixes #97657 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/network/netpol/probe.go [hunk 1]
```diff
@@ -90,27 +88,22 @@ func probeWorker(k8s *kubeManager, jobs <-chan *ProbeJob, results chan<- *ProbeJ
...
-			connected, command, err := k8s.probeConnectivity(podFrom.Namespace, podFrom.Name, containerFrom.Name(), job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain), job.Protocol, job.ToPort)
-			result := &ProbeJobResults{
-				Job:         job,
-				IsConnected: connected,
-				Err:         err,
-				Command:     command,
-			}
-			results <- result
+
+		connected, command, err := k8s.probeConnectivity(
```
Comment (mattfenwick) [inline test/e2e/network/netpol/probe.go:92]: not sure if this was meant to be reformatted or not, I think generally it's done in one line in kube !

File: test/e2e/network/netpol/probe.go [hunk 2]
```diff
@@ -90,27 +88,22 @@ func probeWorker(k8s *kubeManager, jobs <-chan *ProbeJob, results chan<- *ProbeJ
 	defer ginkgo.GinkgoRecover()
 	for job := range jobs {
 		podFrom := job.PodFrom
-		containerFrom, err := podFrom.FindContainer(int32(job.FromPort), job.Protocol)
```
Comment (mattfenwick) [inline test/e2e/network/netpol/probe.go:93]: Is this the only place `FindContainer` was used ?  If so, probably best to remove it.

File: test/e2e/network/netpol/probe.go [hunk 3]
```diff
@@ -90,27 +88,22 @@ func probeWorker(k8s *kubeManager, jobs <-chan *ProbeJob, results chan<- *ProbeJ
...
-		// 1) sanity check that the pod container is found before we run the real test.
-		if err != nil {
```
Comment (mattfenwick) [inline test/e2e/network/netpol/probe.go:95]: 👍 nice one less error case to worry about now!

**Review by mattfenwick — CHANGES_REQUESTED** [review]
Looking good -- couple of changes, squash the commits and I'll lgtm it!

**Comment by jayunit100** [comment]
this looks ok but we just had a major regression ... see #98341...  so lets hold on it for a few days while tests restabilize, then look at it sunday after looking at testgrid  ?

**Comment by knabben** [comment]
tests stabilized, nice catch.

/lgtm

**Pull request merged**